kudu/src/main/java/org/apache/apex/malhar/kudu/BaseKuduOutputOperator.java

```diff
@@ -53,7 +52,7 @@
  */
 public class BaseKuduOutputOperator extends AbstractKuduOutputOperator
 {
-  public static final String DEFAULT_CONNECTION_PROPS_FILE_NAME = "kuduoutputoperator.properties";
+  public static String DEFAULT_CONNECTION_PROPS_FILE_NAME = "kuduoutputoperator.properties";
```

Comment:
why not final?

Reply:
This was done primarily to account for flexible unit testing. 

Kudu cluster based tests are being enabled by just passing in the master hosts definition from the command line. The rest of the properties for Kudu cluster are being enabled via properties file definition. So a temp file is being created whenever the kudu cluster master hosts is passed from the command line. Since we are using the temp file creation API ( your review comment below ) , the file name is being controlled by the API as opposed to us locking it down. Hence removed the "final" qualifier.

Reply:
I think it would be better if environment specific parameters are configured outside of the build (you may have different environments that you want to deploy a build to). Unit tests should also be able to set operator properties to override defaults. Can be done as follow-up PR.

Reply:
I will work on this as an iteration. There are a few more things like security that need to be implemented for this module and will take this up as part of that iteration.

In the current state, the environment is flipped by passing in the kudu cluster environment as a System property using the "-Dkudumasterhosts=xxx.yyy.zzz.aaa" approach thereby allowing configurability for just the Kudu cluster. However the table to be used, the Pojo that represents the table etc are being read from a properties file. The unit test config creates the tables and subsequently uses them in the tests. 

Just to understand your comment a bit more, we would like these configs to be picked up from environment variables instead of the System property ?

Reply:
Not a system property, but operator properties. Something like this here: https://github.com/apache/apex-malhar/blob/master/examples/exactly-once/src/test/java/org/apache/apex/examples/exactlyonce/ExactlyOnceFileOutputAppTest.java#L79